src/ApacheOrcDotNet.OptimizedReader/Infrastructure/CompressedData.cs

```diff
@@ -6,6 +6,26 @@ namespace ApacheOrcDotNet.OptimizedReader.Infrastructure
 {
     public static class CompressedData
     {
+        public static byte[] CheckDecompressionBuffer(ReadOnlySpan<byte> inputBuffer, byte[] targetDecompressionBuffer, CompressionKind compressionKind, int maxDecompressedLengthPerChunk)
```

Comment:
I feel like this would make more sense as `int GetRequiredBufferSize()` rather than doing the reallocation in here.  Unless you're going to run with my previous crazy idea of `CompressedData` owning the buffers.

Reply:
Updated to be `GetRequiredBufferSize` and manage the resize in the caller context.